flink-runtime/src/main/java/org/apache/flink/runtime/operators/BatchTask.java

```diff
@@ -1150,8 +1149,8 @@ else if (index < 0 || index >= this.driver.getNumberOfDriverComparators()) {
 	 * @return The string for logging.
 	 */
 	public static String constructLogString(String message, String taskName, AbstractInvokable parent) {
-		return message + ":  " + taskName + " (" + (parent.getEnvironment().getIndexInSubtaskGroup() + 1) +
-				'/' + parent.getEnvironment().getNumberOfSubtasks() + ')';
+		return message + ":  " + taskName + " (" + (parent.getEnvironment().getTaskInfo().getIndexOfThisSubtask() + 1) +
```

Comment:
That can lead to problems in handling of Chained Tasks. For those, the name of chained task, but the index and parallelism of parent is used.